index.bs

```diff
@@ -164,18 +166,48 @@ In order to <dfn>create new anchor object</dfn> from |native origin| and |sessio
     1. Return |anchor|.
 </div>
 
+<div class="algorithm unstable" data-algorithm="restore-anchor-from-frame">
+The {{XRAnchor/requestPersistentHandle()}} method, when invoked on an {{XRAnchor}} |anchor|, MUST run the following steps:
+    1. Let |promise| be [=a new Promise=] in the [=relevant realm=] of this {{XRSystem}}.
+    1. If |anchor|'s [=XRAnchor/deleted=] is <code>true</code>, [=/reject=] |promise| with {{InvalidStateError}}, return |promise|, and abort these steps.
+    1. Let |session| be |anchor|'s [=XRAnchor/session=].
+    1. If |session|'s [=XRSession/map of persistent anchors=] contains the value |anchor|, run the following steps:
+        1. Let |uuid| be the key of the mapping of the |anchor| value on |session|'s [=XRSession/map of persistent anchors=].
+        1. [=/Resolve=] |promise| with |uuid|.
```

Comment:
I think `[NewObject]` attribute [requires](https://www.w3.org/TR/WebIDL-1/#NewObject) us to always create a new object, so returning already existing instance would break that requirement?

Reply:
Yes, step 1 of the algorithm always creates a new object which is what we return from this functin

Reply:
Hmm, I think I may be misunderstanding what the `[NewObject]` does then - my intuition is that if we annotate a `Promise`-returning method with it, then it will always result in a new object that is carried by the promise (if it doesn't reject) - in this case, I thought we always have to create a new DOMString. If that's not what it means, please disregard!

Reply:
I think you're right. I made the change.